felles/src/main/kotlin/no/nav/helsearbeidsgiver/felles/rapidsrivers/FailKanal.kt

```diff
@@ -6,32 +6,30 @@ import no.nav.helse.rapids_rivers.RapidsConnection
 import no.nav.helse.rapids_rivers.River
 import no.nav.helsearbeidsgiver.felles.EventName
 import no.nav.helsearbeidsgiver.felles.Key
+import no.nav.helsearbeidsgiver.felles.rapidsrivers.model.Fail
+import no.nav.helsearbeidsgiver.utils.json.fromJson
+
 // vi kan vurdere å bruke event feltet og dispatche event istedenfor Fail.
 abstract class FailKanal(val rapidsConnection: RapidsConnection) : River.PacketListener {
     abstract val eventName: EventName
 
     init {
-        configure(
-            River(rapidsConnection).apply {
-                validate(accept())
+        River(rapidsConnection).apply {
+            validate { msg ->
+                msg.demand(
+                    Key.FAIL to {
+                        val fail = it.fromJson(Fail.serializer())
+                        require(fail.event == eventName)
```

Comment:
Hmm, litt merkelig å neste event inni fail. Er dette egentlig nødvendig - vi skal vel fortsette å sende Key.EVENT i feilmeldingene i packet? For meg er det mer logisk / bedre konsistens å lese demand eller require på hvert felt med Key.NAVN slik som øvrige, ved første øyekast trodde jeg du hadde utelatt / glemt hele Event.

Reply:
Jeg var litt fram og tilbake på hvor event burde ligge. Jeg fant ut at jeg ville teste å kun ha det nestet inni fail, men kom frem til at det var greit å ha det både på meldingen og inni fail. Har pushet en commit som legger til event på meldingen.